ambry-store/src/test/java/com.github.ambry.store/CuratedLogIndexState.java

```diff
@@ -137,12 +142,40 @@
    * @throws StoreException
    */
   CuratedLogIndexState(boolean isLogSegmented, File tempDir) throws InterruptedException, IOException, StoreException {
+    this(isLogSegmented, tempDir, false, true);
+  }
+
+  /**
+   * Creates state in order to make sure all cases are represented and log-index tests don't need to do any setup
+   * individually. For understanding the created index, please read the source code which is annotated with comments.
+   * <p/>
+   * Also verifies that the state in {@link #referenceIndex} matches the state in the real index (which implicitly
+   * verifies some functionality of {@link PersistentIndex} and behaviour of the {@link Journal} in the index).
+   * @param isLogSegmented {@code true} if segmented. {@code false} otherwise.
+   * @param tempDir the directory where the log and index files should be created.
+   * @param hardDeleteEnabled if {@code true}, hard delete is enabled.
+   * @param initState sets up a diverse set of entries if {@code true}. Leaves the log and index empty if {@code false}.
+   * @throws InterruptedException
+   * @throws IOException
+   * @throws StoreException
+   */
+  CuratedLogIndexState(boolean isLogSegmented, File tempDir, boolean hardDeleteEnabled, boolean initState)
+      throws InterruptedException, IOException, StoreException {
     this.tempDir = tempDir;
     tempDirStr = tempDir.getAbsolutePath();
     long segmentCapacity = isLogSegmented ? CuratedLogIndexState.SEGMENT_CAPACITY : CuratedLogIndexState.LOG_CAPACITY;
     StoreMetrics metrics = new StoreMetrics(tempDirStr, metricRegistry);
     log = new Log(tempDirStr, CuratedLogIndexState.LOG_CAPACITY, segmentCapacity, metrics);
-    setupTestState(isLogSegmented, segmentCapacity, metrics);
+    metricRegistry = new MetricRegistry();
+    properties.put("store.index.max.number.of.inmem.elements",
+        Integer.toString(CuratedLogIndexState.MAX_IN_MEM_ELEMENTS));
+    properties.put("store.enable.hard.delete", Boolean.toString(hardDeleteEnabled));
+    // not used but set anyway since this is a package private variable.
```

Comment:
`properties` is a package private variable so it is fair for a user of the state to examine the properties if required. So it should reflect the actual config values used to construct the state.